crest/Assets/Crest/Crest/Shaders/OceanInputs/Resources/ClipSurfaceConvexHull.shader

```diff
@@ -25,6 +25,10 @@ Shader "Crest/Inputs/Clip Surface/Convex Hull"
 			#include "../../OceanHelpersNew.hlsl"
 			#include "../../OceanHelpersDriven.hlsl"
 
+			CBUFFER_START(CrestPerOceanInput)
+			uint _DisplacementSamplingIterations;
```

Comment:
Side note - this will stop the loop getting unrolled and potentially make the iteration more expensive. However, how much more expensive depends on hardware and i don't think the clip data is going to render tons of pixels (the lods are modest in resolution and number), and if you feel this is a useful control then lets keep it.

Reply:
I see. Thanks. I am happy to have it reverted. The `uint` on the C# property is enough.

Reply:
I completely misinterpreted that. Would be good to keep it. Sometimes a value of 10 greatly improves the hole.

Reply:
I completely misinterpreted that. Would be good to keep it. Sometimes a value of 10 greatly improves the hole.

Reply:
Ah fair let's keep it then

Reply:
On further testing, ten is only for very extreme waves which I don't think anyone would use since they look broken.

We could replace it with an enum (DisplacementSamplingQuality: Low = 0, Medium = 4, High = 8)? Would duplicating the loop and selecting the right one with an if-else be a better alternative?

Reply:
Yeah maybe that is a good shout. Its hard to know if it will be faster but it seems like a reasonable approach.